ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Rasf/RasfParser.c

```diff
@@ -0,0 +1,103 @@
+/** @file
+  RASF table parser
+
+  Copyright (c) 2024, Arm Limited. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  @par Reference(s):
+    - ACPI 6.5 Specification - August 2022
+**/
+
+#include <Library/PrintLib.h>
+#include <Library/BaseLib.h>
+#include <Library/UefiLib.h>
+#include "AcpiParser.h"
+#include "AcpiView.h"
+
+STATIC ACPI_DESCRIPTION_HEADER_INFO  AcpiHdrInfo;
+
+/**
+  This function is intended to dump out the PCC Id in the RASF Table
+
+  @param [in] Ptr     Pointer to the start of the PCC Id buffer
+  @param [in] Length  The length of the buffer (should be 12)
+**/
+STATIC
+VOID
+DumpPCCId (

```

Comment:
Maybe a `RASF Platform Communication Channel Identifier` field could be added to the `RasfParser` instead. The format could be specified in a similar way as in:
ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Srat/SratParser.c
```
  { L"Reserved",       12, 4, L"%x %x %x %x - %x %x %x %x - %x %x %x %x", Dump12Chars,
```

Reply:
So yeah - that's the problem. ACPI spec on this PCC ID is broken. it has no definition of the contents. I've looked around and it seems it may only be the 4 bytes at the start are needed but these patches to Linux to handle RASF seem a bit broken to me, so I just dumped it out. I think there is a spec issue here, but yes - until that's improved, maybe SratParser would work. It'll save a custom function for that. I did find Dump12Chars but at first glance it did not seem very useful. Here it does.

Reply:
So doing the above would also solve the typo higher up too, but here's the catch. The output of dump12chars() doesn't dump the hex as 2 digits which IMHO it should. i.e. it should say:

    00 01 3f 02 - ff ff 01 00 - 00 00 00 00

not

    0 1 3f 02 - ff ff 1 0 - 0 0 0 0

Which is what it does now. Should I change the dump12chars() to be IMHO nicer, or have an alternative dump12chars() with 2 digit hex?